spec/factories/hyrax_work.rb

```diff
@@ -41,6 +42,15 @@
           .new(resource: work)
           .assign_access_for(visibility: evaluator.visibility_setting)
       end
+      if evaluator.uploaded_files.present?
+        Hyrax::WorkUploadsHandler.new(work: work).add(files: evaluator.uploaded_files).attach
+        evaluator.uploaded_files.each do |file|
+          allow(Hyrax.config.characterization_service).to receive(:run).and_return(true)
+          # I don't love this - we might want to just run background jobs so
+          # this is more real, but we'd have to stub some things.
```

Comment:
could we do

```ruby
ActiveJob::Base.queue_adapter.filter += [ValkyrieIngestJob]
```

Hyrax already makes use of the `ActiveJob` test adapter. i guess the real implementation is probably a bit fussy, since we need to reach into config and then reset it to be really clean:

```ruby
old_perform_enqueued_setting = ActiveJob::Base.queue_adapter.perform_enqueued_jobs
old_job_filter = ActiveJob::Base.queue_adapter.filter 
ActiveJob::Base.queue_adapter.perform_enqueued_jobs = true
ActiveJob::Base.queue_adapter.filter += [ValkyrieIngestJob]

allow(Hyrax.config.characterization_service).to receive(:run).and_return(true)
Hyrax::WorkUploadsHandler.new(work: work).add(files: evaluator.uploaded_files).attach

ActiveJob::Base.queue_adapter.perform_enqueued_jobs = old_perform_enqueued_setting
ActiveJob::Base.queue_adapter.filter = old_job_filter
```

maybe more than is worth messing with?

---

the stub on `Hyrax.config.characterization_service` feels like a code smell to me. currently, if i publish `file.uploaded` from any synchronous code, i'll trigger a long running `characterization_service.run` inline. since any app can publish that event from any code, that might not be the best.

this issue is probably out of scope here, but fixing it would let us rely on the `ActiveJob` adapter filter for the characterization behavior too, which would offer a lot of flexiblitiy.

Reply:
I had a version that did `perform_enqueued jobs`

The big problem is just fits tries to run and it's not installed.

Reply:
does the stub not take effect through the adapter?